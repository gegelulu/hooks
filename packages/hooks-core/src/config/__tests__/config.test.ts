import { join } from 'upath'

import { getConfig, PRE_DEFINE_PROJECT_CONFIG, setConfig } from '../'

beforeEach(() => {
  globalThis[PRE_DEFINE_PROJECT_CONFIG] = undefined
})

test('load pre defined config', async () => {
  setConfig({
    source: '/',
    routes: [{ baseDir: 'lambda', baseRoute: '/api' }],
  })

  expect(getConfig().source).toEqual('/')
})

test('load user config', () => {
  const fixtures = ['js', 'ts', 'ts-export-default']
  fixtures.forEach((fixture) => {
    expect(loadConfig(fixture)).toMatchSnapshot()
  })
})

function loadConfig(fixture: string) {
  return getConfig(join(__dirname, 'fixtures', fixture))
}
