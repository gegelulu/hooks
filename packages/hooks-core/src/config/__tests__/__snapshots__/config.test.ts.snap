// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load user config 1`] = `
Object {
  "build": Object {
    "outDir": "dist",
    "viteOutDir": "build",
  },
  "dev": Object {
    "ignorePattern": [Function],
  },
  "gateway": Array [],
  "request": Object {
    "client": "@midwayjs/hooks-core/request",
  },
  "routes": Array [
    Object {
      "baseDir": "lambda",
      "basePath": "/api",
    },
  ],
  "source": "src",
  "superjson": false,
}
`;

exports[`load user config 2`] = `
Object {
  "build": Object {
    "outDir": "dist",
    "viteOutDir": "build",
  },
  "dev": Object {
    "ignorePattern": [Function],
  },
  "gateway": Array [],
  "request": Object {
    "client": "@midwayjs/hooks-core/request",
  },
  "routes": Array [
    Object {
      "baseDir": "lambda",
      "basePath": "/api",
    },
  ],
  "source": "src",
  "superjson": false,
}
`;

exports[`load user config 3`] = `
Object {
  "build": Object {
    "outDir": "dist",
    "viteOutDir": "build",
  },
  "dev": Object {
    "ignorePattern": [Function],
  },
  "gateway": Array [],
  "request": Object {
    "client": "@midwayjs/hooks-core/request",
  },
  "routes": Array [
    Object {
      "baseDir": "lambda",
      "basePath": "/api",
    },
  ],
  "source": "src",
  "superjson": false,
}
`;

exports[`load user config 4`] = `
Object {
  "build": Object {
    "outDir": "dist",
    "viteOutDir": "build",
  },
  "dev": Object {
    "ignorePattern": [Function],
  },
  "gateway": Array [],
  "presets": Array [
    [Function],
  ],
  "request": Object {
    "client": "@TEST/@midwayjs/hooks-core/request",
  },
  "routes": Array [
    Object {
      "baseDir": "lambda",
      "basePath": "/preset-api",
    },
  ],
  "source": "src",
  "superjson": false,
}
`;

exports[`load user config 5`] = `
Object {
  "build": Object {
    "outDir": "dist",
    "viteOutDir": "build",
  },
  "dev": Object {
    "ignorePattern": [Function],
  },
  "gateway": Array [],
  "presets": Array [
    [Function],
    [Function],
    [Function],
  ],
  "request": Object {
    "client": "@TEST/@midwayjs/hooks-core/request",
  },
  "routes": Array [
    Object {
      "baseDir": "lambda",
      "basePath": "/preset-api",
    },
    Object {
      "baseDir": "lambda-second",
      "basePath": "/second-preset-api",
    },
    Object {
      "baseDir": "lambda-second",
      "basePath": "/third-preset-api",
    },
  ],
  "source": "src",
  "superjson": false,
}
`;
